### ISSUE TITLE
InRegionS3PolicyLambdaFunction | Unzipped size must be smaller than 262144000 bytes

### ISSUE THREAD

**Issue by garysieling** [nasa/cumulus#1021]
When I deploy Cumulus v1.13.0, I'm getting an error that one of the lambdas is too large unzipped:

InRegionS3PolicyLambdaFunction | Unzipped size must be smaller than 262144000 bytes (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 50d92946-814b-11e9-bbaf-9f7f6002a11b)

S3 shows the file as 54.9 MB. When I extract it, it's 319 mb. Within the zip file, there are the following folders:

 49M	app
 25M	bootstrap
 12K	bulkDelete
 14M	cleanExecutions
 16M	createReconciliationReport
 25M	dbIndexer
 27M	distribution
5.7M	emsDistributionReport
 26M	emsReport
 25M	executeMigrations
 25M	indexer
 25M	jobs
 14M	messageConsumer
4.1M	payloadLogger
 14M	sfScheduler
4.2M	sfSemaphoreDown
1.6M	sfSnsBroadcast
4.2M	sfStarter

It looks like the space is consumed by unminified output of webpack. All these files appear to come from the NPM distribution, so I'm curious how this works for other people.

Is this a known issue? I'll update this ticket once I find out more information.

**Comment by garysieling** [comment]
I was able to upgrade by removing this lambda entirely:

{code}
InRegionS3Policy:
  handler: index.inRegionS3Policy
  memory: 256
  timeout: 300
  source: node_modules/@cumulus/api/dist/
{code}

Is it possible this configuration is wrong (i.e. sucking in too much javascript)?

**Comment by garysieling** [comment]
I was able to work around this by minifying the Cumulus lambdas:
```

echo "Minifying Cumulus"
cd ../cumulus-deploy
npm install -g terser 
for i in ${api_lambdas[@]}; do   
    echo $i;   
    ls -al $i;     
    terser $i > temp.js;     
    rm $i;     
    mv temp.js $i;     
    ls -al $i; 
done; 
cd ../build-deploy```

**Comment by laurenfrederick** [comment]
@garysieling This should all be fixed in 1.13. We removed the `InRegionS3Policy` lambda all together and made fixes to the lambdas to only pull in the code needed.

**Comment by garysieling** [comment]
Ok, thanks, I'll just remove the reference from our deployment then.
